MapboxNavigation/NavigationMapView.swift

```diff
@@ -764,6 +767,49 @@ open class NavigationMapView: MGLMapView {
             style.removeSource(arrowSymboleSource)
         }
     }
+    
+    public func showVoiceInstructionsOnMap(route: Route) {
+        guard let style = style else {
+            return
+        }
+        
+        var features = [MGLPointFeature]()
+        for (legIndex, leg) in route.legs.enumerated() {
+            for (stepIndex, step) in leg.steps.enumerated() {
+                for instruction in step.instructionsSpokenAlongStep! {
+                    let feature = MGLPointFeature()
+                    feature.coordinate = Polyline(route.legs[legIndex].steps[stepIndex].coordinates!.reversed()).coordinateFromStart(distance: instruction.distanceAlongStep)!
+                    feature.attributes = [ "instruction": instruction.text ]
+                    features.append(feature)
+                }
+            }
+        }
+        
+        let instructionPointSource = MGLShapeCollectionFeature(shapes: features)
+        
+        if let instructionSource = style.source(withIdentifier: instructionSource) as? MGLShapeSource {
+            instructionSource.shape = instructionPointSource
+        } else {
+            let sourceShape = MGLShapeSource(identifier: instructionSource, shape: instructionPointSource, options: nil)
+            style.addSource(sourceShape)
+            
+            let symbol = MGLSymbolStyleLayer(identifier: instructionLabel, source: sourceShape)
+            symbol.text = MGLStyleValue(rawValue: "{instruction}")
+            symbol.textFontSize = MGLStyleValue(rawValue: 14)
+            symbol.textHaloWidth = MGLStyleValue(rawValue: 1)
+            symbol.textHaloColor = MGLStyleValue(rawValue: .white)
+            symbol.textOpacity = MGLStyleValue(rawValue: 0.75)
+            symbol.textAnchor = MGLStyleValue(rawValue: NSValue(mglTextAnchor: .left))
```

Comment:
This isn’t what I meant by https://github.com/mapbox/mapbox-navigation-ios/pull/727#discussion_r145259373. I meant that `bottomLeft` is a fine text anchor, but `textJustify` should be set to `left` as well, instead of the default value of `center`.

Reply:
#758